rust/src/dataset/scanner.rs

```diff
@@ -330,7 +327,7 @@ impl Scanner {
     /// Create an Execution plan to do indexed ANN search
     fn ann(&self, q: &Query, index: &&Index) -> Arc<dyn ExecutionPlan> {
         let mut inner_query = q.clone();
-        inner_query.k = q.k * (q.refine_factor.unwrap_or(1) as usize);
+        inner_query.k = q.k * std::cmp::max(1, q.refine_factor.unwrap_or(1) as usize);
```

Comment:
this is preventing to set refine_factor to 0? Should we detect that earlier during user inputs?